core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -372,6 +378,13 @@ public AuthSession(
       this.credential = credential;
     }
 
+    // this field is publicly visible/configurable for testing only. We can't annotate it with
+    // @VisibleForTesting because we restrict via checkstyle that @VisibleForTesting should be only
+    // applied to package-private things
+    public static void withTokenRefreshNumRetries(int retries) {
```

Comment:
Rather than making this public, can we add a public method in a test util that sets it? That way we could make it package-private with a helper class.

Reply:
done